### PR TITLE
Extend import puppetclasses test to cover BZ 2142555

### PR DESCRIPTION
**Pull request by Gauravtalreja1**

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_smartproxy.py -k test_positive_import_puppet_classes

File: tests/foreman/api/test_smartproxy.py [hunk 1]
```diff
@@ -233,29 +234,66 @@ def test_positive_refresh_features(request, target_sat):
...
+        organization=[module_puppet_org],
+        location=[module_puppet_loc],
+    ).create()
+    request.addfinalizer(user.delete)
+    request.addfinalizer(role.delete)
+
+    new_port = puppet_sat.available_capsule_port
+    with puppet_sat.default_url_on_new_port(9090, new_port) as url:
+        proxy = puppet_sat.api.SmartProxy(url=url).create()
+        result = proxy.import_puppetclasses()
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_smartproxy.py:283]: for non-admin you are passing `enviornment` why this is the case for admin user ?

**Review by omkarkhatavkar — APPROVED** [review]
ACK pending question

File: tests/foreman/api/test_smartproxy.py (hunk 1 above)
Comment (vijaysawant) [reply tests/foreman/api/test_smartproxy.py:283]: I think, passing the environment to the non-admin user is for adding automation coverage mentioned in BZ[#2142555](https://bugzilla.redhat.com/show_bug.cgi?id=2142555), making API endpoint.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_smartproxy.py -k test_positive_import_puppet_classes

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_smartproxy.py -k test_positive_import_puppet_classes

File: tests/foreman/api/test_smartproxy.py [hunk 2]
```diff
@@ -233,29 +234,66 @@ def test_positive_refresh_features(request, target_sat):
...
+    """Import puppet classes from proxy for admin and non-admin user
 
     :id: 385efd1b-6146-47bf-babf-0127ce5955ed
 
     :expectedresults: Puppet classes are imported from proxy
 
     :CaseLevel: Integration
 
-    :BZ: 1398695
+    :BZ: 1398695, 2142555
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_smartproxy.py:252]: updated

File: tests/foreman/api/test_smartproxy.py [hunk 3]
```diff
@@ -233,29 +234,66 @@ def test_positive_refresh_features(request, target_sat):
...
-        new_port = puppet_sat.available_capsule_port
-        with puppet_sat.default_url_on_new_port(9090, new_port) as url:
-            proxy = puppet_sat.api.SmartProxy(url=url).create()
-            result = proxy.import_puppetclasses()
-            assert (
-                "Successfully updated environment and puppetclasses from "
-                "the on-disk puppet installation"
-            ) in result['message'] or "No changes to your environments detected" in result[
-                'message'
+    puppet_sat = session_puppet_enabled_sat
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/api/test_smartproxy.py:256]: yes, its good to keep it short

File: tests/foreman/api/test_smartproxy.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_smartproxy.py:290]: updated test to cover both w/ and w/o environment scenarios to import puppetclasses for both admin and non-admin users, thanks!

**Review by shubhamsg199 — APPROVED** [review]
Ack

**Pull request merged**